middleware/middleware.go

```diff
@@ -37,6 +38,50 @@ func MakeEndpointInstrumentingMW(m metrics.Metrics, histoName string) cs.Middlew
 	}
 }
 
+// IDRetriever is an interface to get an ID using an object's name
+type IDRetriever interface {
+	GetID(accessToken, name string) (string, error)
+}
+
+// AdminConfigurationRetriever is an interface to get an admin configuration
+type AdminConfigurationRetriever interface {
+	GetAdminConfiguration(ctx context.Context, realmID string) (configuration.RealmAdminConfiguration, error)
+}
+
+// MakeEndpointAvailableCheckMW makes a middleware that ensure a feature is enabled at admin configuration level in the current context
+func MakeEndpointAvailableCheckMW(realmKey, enabledKey string, idRetriever IDRetriever, confRetriever AdminConfigurationRetriever, logger log.Logger) func(http.Handler) http.Handler {
```

Comment:
realmKey seems to be not used

Reply:
Yes... It was used in a previous version but not needed anymore.
I'm fixing it